libs/net/dal/Services/DataSourceService.cs

```diff
@@ -74,12 +74,38 @@ public IPaged<DataSource> Find(DataSourceFilter filter)
             .Include(c => c.DataLocation)
             .Include(c => c.MediaType)
             .Include(c => c.License)
+            .Include(c => c.Parent)
             .Include(c => c.ActionsManyToMany).ThenInclude(ca => ca.SourceAction)
             .Include(c => c.MetricsManyToMany).ThenInclude(cc => cc.SourceMetric)
             .Include(c => c.SchedulesManyToMany).ThenInclude(ct => ct.Schedule)
             .FirstOrDefault(c => c.Id == id);
     }
 
+    public DataSource? FindByCode(string code)
+    {
+        return this.Context.DataSources
+            .Include(c => c.DataLocation)
+            .Include(c => c.MediaType)
+            .Include(c => c.License)
+            .Include(c => c.Parent)
```

Comment:
Added Parent when asking for a single data-source.  This is to support the clip service.  I'd prefer to not globally include this, but for now it should work.